src/types.rs

```diff
@@ -0,0 +1,5 @@
+#![allow(non_camel_case_types)]
+pub type c_int = i32;
+pub type c_uchar = u8;
+pub type c_uint = u32;
+pub type c_void = c_uchar;
```

Comment:
Is it really correct to just use `u8` for `c_void`? That comment makes me think otherwise https://doc.rust-lang.org/src/core/ffi.rs.html#21 (though I don't understand it).

Reply:
In one of the comments here I think @apoelstra  pointed out that *void is almost equivalent to *char, it's hard to look for resolved comments on the phone.

Reply:
But wouldn't it be cleaner to copy the enum implementation with the two variants? It's just three lines of code, so it wouldn't hurt I guess.

Reply:
Agree with real-or-random. I don't mind type-punning `*c_void` for `*c_uchar` but the base types `c_void` and `c_uchar` are very different.

Reply:
So I see two options.
one is to remove the `c_void` type and replace the pointers to `*c_uchar`, another is what @real-or-random suggested and copy the enum implementation.

I'm really no expert in C and don't know the implications if someone later on misuse `c_void` with `c_uchar` so I don't have any opinion on this, I think it should be as foolproof as possible.

Reply:
I think we should copy the enum implementation and link to the libc code, and mention that in Rust 1.30 we will be able to just use `core::ffi`.

Reply:
Ok, FYI `c_void != void`.
but `*const c_void == const void* && *mut c_void == void*`